tracer/src/Datadog.Trace/Util/TypeExtensions.cs

```diff
@@ -0,0 +1,21 @@
+// <copyright file="TypeExtensions.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System.Reflection;
+using System.Runtime.CompilerServices;
+
+namespace Datadog.Trace.Util;
+
+internal static class TypeExtensions
+{
+    public static bool IsAnonymous(this TypeInfo type)
+    {
+        return type.GetCustomAttribute<CompilerGeneratedAttribute>() != null
+            && type.IsGenericType && type.Name.Contains("AnonymousType")
+            && (type.Name.StartsWith("<>") || type.Name.StartsWith("VB$"))
+            && (type.Attributes & TypeAttributes.NotPublic) == TypeAttributes.NotPublic;
```

Comment:
`TypeAttributes.NotPublic == 0`, which means that last check is _always_ true, because it always evaluates to `0 == 0`...

I think you would instead need to do 
```suggestion
            && (type.Attributes & TypeAttributes.Public) != TypeAttributes.Public;
```

Reply:
yes good catch!! actually it's funny cause I used it from: https://github.com/DataDog/dd-trace-dotnet/blob/43efa46a49669568b3bd2d6aca6675ae03be4d19/tracer/src/Datadog.Trace/Vendors/MessagePack/Internal/ReflectionExtensions.cs#L33 , but changed it in https://github.com/DataDog/dd-trace-dotnet/pull/4865/commits/2725680c77513375b1d3b496c8f6e9cb4cfe3b23